controllers/storagecluster/external_resources_test.go

```diff
@@ -44,7 +44,8 @@ var globalTestExternalResources = []ExternalResource{
 	{
 		Kind: "StorageClass",
 		Data: map[string]string{
-			"pool": "device_health_metrics",
+			"pool":     "device_health_metrics",
```

Comment:
We need a specific test for EC and non-EC. This is just adding EC and not validating it. Also, it removes the non-EC case.

Reply:
added unit test.